add pitch recipe for nnet2/3 online setup

naxingyu opened the pull request:
This is for addressing #1011

danpovey commented:
How much better is this than without pitch?
Or you had pitch before, but just not for the nnet3 setup?

naxingyu commented:
The hkust/s5 recipe was w/o pitch. That’s why I started s5b. Improvement can be seen by comparing the RESULTS files.

> 在 2016年12月21日，下午1:03，Daniel Povey <notifications@github.com> 写道：
> 
> How much better is this than without pitch?
> Or you had pitch before, but just not for the nnet3 setup?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-268436150>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ADKpxMjovZt2BmJ9Q0NuFEt1b2hj2BBfks5rKLMJgaJpZM4LSi8m>.
>

danpovey commented on egs/hkust/s5b/local/create_oov_char_lexicon.pl:
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/perl
```
shebang should be e.g. /usr/bin/env perl

danpovey commented on egs/hkust/s5b/local/hkust_data_prep.sh:
```diff
@@ -0,0 +1,167 @@
+#!/bin/bash
+
+
+. path.sh
```
use pathname, and check it's there i.e.
`. ./path.sh || exit 1`

danpovey replied:
or alternatively, do
set -e
above this, which may help in error checking in the rest of the file (you don't seem to be doing '|| exit 1' anywhere).

danpovey commented on egs/hkust/s5b/local/hkust_data_prep.sh:
```diff
@@ -0,0 +1,167 @@
+. path.sh
+
+if [ $# != 2 ]; then
+   echo "Usage: hkust_data_prep.sh AUDIO_PATH TEXT_PATH"
```
change message to be in this style:
```
Usage: $0 <audio-path> <text-path>
e.g.: $0 /export/corpora/LDC43142B /export/corpora/LDC43142X
```
... obviously use the correct directories, that look plausible (maybe it's not LDC).

danpovey commented on egs/hkust/s5b/local/hkust_data_prep.sh:
```diff
@@ -0,0 +1,167 @@
+fi
+
+HKUST_AUDIO_DIR=$1
+HKUST_TEXT_DIR=$2
```
please avoid uppercase variable names.

danpovey commented on egs/hkust/s5b/local/hkust_data_prep.sh:
```diff
@@ -0,0 +1,167 @@
+cat $dev_dir/segments | awk '{spk=substr($1,1,33); print $1 " " spk}' > $dev_dir/utt2spk || exit 1;
+cat $dev_dir/utt2spk | sort -k 2 | utils/utt2spk_to_spk2utt.pl > $dev_dir/spk2utt || exit 1;
+
+echo HKUST data preparation succeeded
```
`echo "$0: HKUST data preparation succeeded`

danpovey commented on egs/hkust/s5b/local/hkust_format_data.sh:
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+#
+
+if [ -f path.sh ]; then . path.sh; fi
```
again, should be ./path.sh.

danpovey commented on egs/wsj/s5/utils/data/limit_feature_dim.sh:
```diff
@@ -0,0 +1,43 @@
+
+if [ $# != 3 ]; then
+  echo "Usage: "
+  echo "  $0 <featdim> <srcdir> <destdir>"
```
featdim -> feature-dim-range

danpovey commented on egs/hkust/s5b/run.sh:
```diff
@@ -0,0 +1,162 @@
+  --transform-dir exp/tri5a/decode \
+  exp/tri5a/graph data/dev exp/tri5a_mpe/decode || exit 1 ;
+
+# SGMM system [sgmm5a]
```
If you have time, I think it would be better to move the parts of the script that relate to discriminative training and SGMMs, to sub-scripts in local, e.g.
local/run_discriminative.sh
and
local/run_sgmm.sh

and put those commands commented-out in the top-level run.sh, since those aren't the branches of the system building that we particularly recommend at this point.

danpovey commented on egs/hkust/s5b/local/hkust_segment.py:
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env python
+#coding:utf-8
+#!/usr/bin/env python
```
remove extra shebang

danpovey commented on egs/hkust/s5b/local/hkust_prepare_dict.sh:
```diff
@@ -0,0 +1,304 @@
+gawk 'NR==FNR{words[$1]; next;} ($1 in words)' \
+  $dict_dir/map/cmu-not-used conf/cmu2pinyin |\
+  egrep -v '<.?s>' > $dict_dir/map/cmu-py
+
```
some comments throughout this script would be nice.

danpovey commented on egs/hkust/s5b/local/hkust_prepare_dict.sh:
```diff
@@ -0,0 +1,304 @@
+  'BEGIN{while((getline<cmu)) dict[$1] = 1;}
+   {if (!dict[$1]) print $1;}' > $dict_dir/map/cmu-not-used
+
+gawk 'NR==FNR{words[$1]; next;} ($1 in words)' \
```
Can you do this in a way that doesn't require gawk?  In general we are avoiding the dependency on gawk, unless as required by SRILM.

danpovey commented on egs/hkust/s5b/local/hkust_prepare_dict.sh:
```diff
@@ -0,0 +1,304 @@
+  ' >  $dict_dir/lexicon-ch/ch-char-dict-mp.txt
+
+# get lexicon for Chinese OOV words
+perl local/create_oov_char_lexicon.pl $dict_dir/lexicon-ch/ch-char-dict-mp.txt $dict_dir/lexicon-ch/words-ch-oov.txt > $dict_dir/lexicon-ch/lexicon-ch-oov.txt
```
don't invoke with perl, rely on the shebang.

danpovey commented on egs/hkust/s5b/local/hkust_data_prep.sh:
```diff
@@ -0,0 +1,167 @@
+  python setup.py build 
+  python setup.py install --prefix=.
+  cd ../..
+  if [ ! -d tools/mmseg-1.3.0/lib/python${pyver}/site-packages ]; then
```
In general I don't like automatically installing stuff without requiring user interaction, because it's a security risk.   What I would be happier with is adding a script to install this like tools/extras/install_mmseg.sh, and (maybe near the top of run.sh as well as inside this script) prompting the user to invoke that if it's not installed.  But if you really don't have time for this, I'd consider just adding  warning in a comment near the top of run.sh that mentions that this is automatically installed.

danpovey commented on egs/hkust/s5b/local/score.sh:
```diff
@@ -0,0 +1,108 @@
+  [ ! -f $f ] && echo "$0: expecting file $f to exist" && exit 1;
+done
+
+name=`basename $data`; # e.g. eval2000
```
please change eval2000 to something relevant for this datset.

naxingyu commented:
AFAIK, there are some groups in China who have built the old hkust system. But it’s OK if you want it as a replacement to s5.

> 在 2016年12月26日，09:47，Daniel Povey <notifications@github.com> 写道：
> 
> @danpovey commented on this pull request.
> 
> A major change I am considering asking you to make is to just replace the old hkust/s5 recipe with this recipe.
> My feeling is that the number of people who have built systems with the old recipe may be small enough that we don't have to worry too much about causing some back-compatibility issues for them (i.e. the added bulk in the repo size would be more a problem, because it affects everyone).
> @jtrmal <https://github.com/jtrmal>, do you have any opinions on this?
> 
> In egs/wsj/s5/utils/data/limit_feature_dim.sh <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14370753>:
> 
> > @@ -0,0 +1,43 @@
> +#!/bin/bash
> +
> +# Copyright 2016  Alibaba Robotics Corp. (author: Xingyu Na)
> +# Apache 2.0
> +
> +. utils/parse_options.sh
> +
> +if [ $# != 3 ]; then
> +  echo "Usage: "
> +  echo "  $0 <featdim> <srcdir> <destdir>"
> featdim -> feature-dim-range
> 
> In egs/hkust/s5b/run.sh <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14370753>:
> 
> > +  --config conf/decode.config \
> +  data/train data/lang exp/tri5a exp/tri5a_denlats || exit 1;
> +steps/train_mmi.sh --cmd "$train_cmd" --boost 0.1 \
> +  data/train data/lang exp/tri5a_ali exp/tri5a_denlats exp/tri5a_mmi_b0.1 || exit 1;
> +steps/decode.sh --cmd "$decode_cmd" --nj 10 --config conf/decode.config \
> +  --transform-dir exp/tri5a/decode \
> +  exp/tri5a/graph data/dev exp/tri5a_mmi_b0.1/decode || exit 1 ;
> +
> +# Do MPE.
> +steps/train_mpe.sh  --cmd "$train_cmd" data/train data/lang exp/tri5a_ali exp/tri5a_denlats exp/tri5a_mpe || exit 1;
> +
> +steps/decode.sh --cmd "$decode_cmd" --nj 10 --config conf/decode.config \
> +  --transform-dir exp/tri5a/decode \
> +  exp/tri5a/graph data/dev exp/tri5a_mpe/decode || exit 1 ;
> +
> +# SGMM system [sgmm5a]
> If you have time, I think it would be better to move the parts of the script that relate to discriminative training and SGMMs, to sub-scripts in local, e.g.
> local/run_discriminative.sh
> and
> local/run_sgmm.sh
> 
> and put those commands commented-out in the top-level run.sh, since those aren't the branches of the system building that we particularly recommend at this point.
> 
> In egs/hkust/s5b/local/hkust_segment.py <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14370753>:
> 
> > @@ -0,0 +1,15 @@
> +#!/usr/bin/env python
> +#coding:utf-8
> +#!/usr/bin/env python
> remove extra shebang
> 
> In egs/hkust/s5b/local/hkust_prepare_dict.sh <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14370753>:
> 
> > +cat $dict_dir/lexicon-en/lexicon-en-oov.txt $dict_dir/lexicon-en/lexicon-en-iv.txt |\
> +  sort > $dict_dir/lexicon-en/lexicon-en-phn.txt
> +
> +mkdir $dict_dir/map
> +cat conf/cmu2pinyin | awk '{print $1;}' | sort -u > $dict_dir/map/cmu
> +cat conf/pinyin2cmu | awk -v cmu=$dict_dir/map/cmu \
> +  'BEGIN{while((getline<cmu)) dict[$1] = 1;}
> +   {for (i = 2; i <=NF; i++) if (dict[$i]) print $i;}' | sort -u > $dict_dir/map/cmu-used
> +cat $dict_dir/map/cmu | awk -v cmu=$dict_dir/map/cmu-used \
> +  'BEGIN{while((getline<cmu)) dict[$1] = 1;}
> +   {if (!dict[$1]) print $1;}' > $dict_dir/map/cmu-not-used
> +
> +gawk 'NR==FNR{words[$1]; next;} ($1 in words)' \
> +  $dict_dir/map/cmu-not-used conf/cmu2pinyin |\
> +  egrep -v '<.?s>' > $dict_dir/map/cmu-py
> +
> some comments throughout this script would be nice.
> 
> In egs/hkust/s5b/local/hkust_prepare_dict.sh <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14370753>:
> 
> > +g2p.py --model=conf/g2p_model --apply $dict_dir/lexicon-en/words-en-oov.txt \
> +  > $dict_dir/lexicon-en/lexicon-en-oov.txt
> +
> +cat $dict_dir/lexicon-en/lexicon-en-oov.txt $dict_dir/lexicon-en/lexicon-en-iv.txt |\
> +  sort > $dict_dir/lexicon-en/lexicon-en-phn.txt
> +
> +mkdir $dict_dir/map
> +cat conf/cmu2pinyin | awk '{print $1;}' | sort -u > $dict_dir/map/cmu
> +cat conf/pinyin2cmu | awk -v cmu=$dict_dir/map/cmu \
> +  'BEGIN{while((getline<cmu)) dict[$1] = 1;}
> +   {for (i = 2; i <=NF; i++) if (dict[$i]) print $i;}' | sort -u > $dict_dir/map/cmu-used
> +cat $dict_dir/map/cmu | awk -v cmu=$dict_dir/map/cmu-used \
> +  'BEGIN{while((getline<cmu)) dict[$1] = 1;}
> +   {if (!dict[$1]) print $1;}' > $dict_dir/map/cmu-not-used
> +
> +gawk 'NR==FNR{words[$1]; next;} ($1 in words)' \
> Can you do this in a way that doesn't require gawk? In general we are avoiding the dependency on gawk, unless as required by SRILM.
> 
> In egs/hkust/s5b/local/hkust_prepare_dict.sh <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14370753>:
> 
> > +  my $out_line = "";
> +  while (<STDIN>) {
> +    @A = split(" ", $_);
> +    $cur = $A[0];
> +    $cur_py = $A[1];
> +    #print length($prev);
> +    if (length($prev) == 0) { $out_line = $_; chomp($out_line);}
> +    if (length($prev)>0 && $cur ne $prev) { print $out_line; print "\n"; $out_line = $_; chomp($out_line);}
> +    if (length($prev)>0 && $cur eq $prev) { $out_line = $out_line."/"."$cur_py";}
> +    $prev = $cur;
> +  }
> +  print $out_line;
> +  ' >  $dict_dir/lexicon-ch/ch-char-dict-mp.txt
> +
> +# get lexicon for Chinese OOV words
> +perl local/create_oov_char_lexicon.pl $dict_dir/lexicon-ch/ch-char-dict-mp.txt $dict_dir/lexicon-ch/words-ch-oov.txt > $dict_dir/lexicon-ch/lexicon-ch-oov.txt
> don't invoke with perl, rely on the shebang.
> 
> In egs/hkust/s5b/local/hkust_data_prep.sh <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14370753>:
> 
> > +#transcripts normalization and segmentation 
> +#(this needs external tools),
> +#Download and configure segment tools   
> +pyver=`python --version 2>&1 | sed -e 's:.*\([2-3]\.[0-9]\+\).*:\1:g'`
> +export PYTHONPATH=$PYTHONPATH:`pwd`/tools/mmseg-1.3.0/lib/python${pyver}/site-packages
> +if [ ! -d tools/mmseg-1.3.0/lib/python${pyver}/site-packages ]; then
> +  echo "--- Downloading mmseg-1.3.0 ..."
> +  echo "NOTE: it assumes that you have Python, Setuptools installed on your system!"
> +  wget -P tools http://pypi.python.org/packages/source/m/mmseg/mmseg-1.3.0.tar.gz 
> +  tar xf tools/mmseg-1.3.0.tar.gz -C tools
> +  cd tools/mmseg-1.3.0
> +  mkdir -p lib/python${pyver}/site-packages
> +  python setup.py build 
> +  python setup.py install --prefix=.
> +  cd ../..
> +  if [ ! -d tools/mmseg-1.3.0/lib/python${pyver}/site-packages ]; then
> In general I don't like automatically installing stuff without requiring user interaction, because it's a security risk. What I would be happier with is adding a script to install this like tools/extras/install_mmseg.sh, and (maybe near the top of run.sh as well as inside this script) prompting the user to invoke that if it's not installed. But if you really don't have time for this, I'd consider just adding warning in a comment near the top of run.sh that mentions that this is automatically installed.
> 
> In egs/hkust/s5b/local/score.sh <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14370753>:
> 
> > +
> +data=$1
> +lang=$2 # Note: may be graph directory not lang directory, but has the necessary stuff copied.
> +dir=$3
> +
> +model=$dir/../$iter.mdl # assume model one level up from decoding dir.
> +
> +hubscr=$KALDI_ROOT/tools/sctk/bin/hubscr.pl
> +[ ! -f $hubscr ] && echo "Cannot find scoring program at $hubscr" && exit 1;
> +hubdir=`dirname $hubscr`
> +
> +for f in $data/text $lang/words.txt $dir/lat.1.gz; do
> +  [ ! -f $f ] && echo "$0: expecting file $f to exist" && exit 1;
> +done
> +
> +name=`basename $data`; # e.g. eval2000
> please change eval2000 to something relevant for this datset.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14370753>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ADKpxP8LXNxDgM-vvb8qLeRUcVyfy3ndks5rLxyXgaJpZM4LSi8m>.
>

naxingyu commented:
@danpovey I've removed s5b and redirect all the changes to s5. All comments are addressed. Chain recipe is added to address #1270

jtrmal commented:
I agree that creating a new recipe is  better option.
What was causing sore eyes in the original recipe was the text data
preprocessing. That script was huge and I had terrible times just trying to
fix things to work for me. And I wasn;t trying to improve it -- just trying
to make it work.
So perhaps it might be beneficial to make the script perhaps more verbose
or structured differently or some of the functionality moved into a
separate script?
y.

On Wed, Dec 28, 2016 at 9:09 AM, Xingyu Na <notifications@github.com> wrote:

> @danpovey <https://github.com/danpovey> I've removed s5b and redirect all
> the changes to s5. All comments are addressed. Chain recipe is added to
> address #1270 <https://github.com/kaldi-asr/kaldi/issues/1270>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269441818>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKisX1PpyoTWPTkmBdI5FZFw1MQrrs0Yks5rMhkwgaJpZM4LSi8m>
> .
>

danpovey commented:
@jtrmal, can you clarify about "creating a new recipe", do you mean moving
to s5b?  If so, what do you think the advantages are of having a new recipe
number?
I'm not sure that I want to encourage too much of this duplication in too
many of the recipes, it will bulk up the checked-out repo.



On Wed, Dec 28, 2016 at 1:16 AM, jtrmal <notifications@github.com> wrote:

> I agree that creating a new recipe is better option.
> What was causing sore eyes in the original recipe was the text data
> preprocessing. That script was huge and I had terrible times just trying to
> fix things to work for me. And I wasn;t trying to improve it -- just trying
> to make it work.
> So perhaps it might be beneficial to make the script perhaps more verbose
> or structured differently or some of the functionality moved into a
> separate script?
> y.
>
> On Wed, Dec 28, 2016 at 9:09 AM, Xingyu Na <notifications@github.com>
> wrote:
>
> > @danpovey <https://github.com/danpovey> I've removed s5b and redirect
> all
> > the changes to s5. All comments are addressed. Chain recipe is added to
> > address #1270 <https://github.com/kaldi-asr/kaldi/issues/1270>
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269441818>,
> or mute
> > the thread
> > <https://github.com/notifications/unsubscribe-auth/
> AKisX1PpyoTWPTkmBdI5FZFw1MQrrs0Yks5rMhkwgaJpZM4LSi8m>
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269449318>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVux9yfp0OR2IyKAlhJ7Md3sGDSf9tks5rMij0gaJpZM4LSi8m>
> .
>

jtrmal commented:
Yes,  I meant creating s5b,  sorry.
I was thinking it might be beneficial because of changes in the scoring
scripts and and data preprocessing,  I. E.   Loss in the
continuity/comparability of the scores.
Y.

On Dec 28, 2016 7:54 PM, "Daniel Povey" <notifications@github.com> wrote:

@jtrmal, can you clarify about "creating a new recipe", do you mean moving
to s5b? If so, what do you think the advantages are of having a new recipe
number?
I'm not sure that I want to encourage too much of this duplication in too
many of the recipes, it will bulk up the checked-out repo.




On Wed, Dec 28, 2016 at 1:16 AM, jtrmal <notifications@github.com> wrote:

> I agree that creating a new recipe is better option.
> What was causing sore eyes in the original recipe was the text data
> preprocessing. That script was huge and I had terrible times just trying
to
> fix things to work for me. And I wasn;t trying to improve it -- just
trying
> to make it work.
> So perhaps it might be beneficial to make the script perhaps more verbose
> or structured differently or some of the functionality moved into a
> separate script?
> y.
>
> On Wed, Dec 28, 2016 at 9:09 AM, Xingyu Na <notifications@github.com>
> wrote:
>
> > @danpovey <https://github.com/danpovey> I've removed s5b and redirect

> all
> > the changes to s5. All comments are addressed. Chain recipe is added to
> > address #1270 <https://github.com/kaldi-asr/kaldi/issues/1270>
> >
> > —
> > You are receiving this because you were mentioned.

> > Reply to this email directly, view it on GitHub
> > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269441818>,

> or mute
> > the thread
> > <https://github.com/notifications/unsubscribe-auth/
> AKisX1PpyoTWPTkmBdI5FZFw1MQrrs0Yks5rMhkwgaJpZM4LSi8m>
> > .
> >
>
> —
> You are receiving this because you were mentioned.

> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269449318>, or
mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/
ADJVux9yfp0OR2IyKAlhJ7Md3sGDSf9tks5rMij0gaJpZM4LSi8m>
> .
>

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269522522>, or mute
the thread
<https://github.com/notifications/unsubscribe-auth/AKisX4ROHXQe6HYxw91cXkA1pPONxzG6ks5rMrB2gaJpZM4LSi8m>
.

danpovey commented:
OK.. Xingyu-- if it's easy to go back to creating a different recipe (s5b)
then go ahead... I think you already changed the PR...


On Wed, Dec 28, 2016 at 11:27 AM, jtrmal <notifications@github.com> wrote:

> Yes, I meant creating s5b, sorry.
> I was thinking it might be beneficial because of changes in the scoring
> scripts and and data preprocessing, I. E. Loss in the
> continuity/comparability of the scores.
> Y.
>
>
> On Dec 28, 2016 7:54 PM, "Daniel Povey" <notifications@github.com> wrote:
>
> @jtrmal, can you clarify about "creating a new recipe", do you mean moving
> to s5b? If so, what do you think the advantages are of having a new recipe
> number?
> I'm not sure that I want to encourage too much of this duplication in too
> many of the recipes, it will bulk up the checked-out repo.
>
>
>
>
> On Wed, Dec 28, 2016 at 1:16 AM, jtrmal <notifications@github.com> wrote:
>
> > I agree that creating a new recipe is better option.
> > What was causing sore eyes in the original recipe was the text data
> > preprocessing. That script was huge and I had terrible times just trying
> to
> > fix things to work for me. And I wasn;t trying to improve it -- just
> trying
> > to make it work.
> > So perhaps it might be beneficial to make the script perhaps more verbose
> > or structured differently or some of the functionality moved into a
> > separate script?
> > y.
> >
> > On Wed, Dec 28, 2016 at 9:09 AM, Xingyu Na <notifications@github.com>
> > wrote:
> >
> > > @danpovey <https://github.com/danpovey> I've removed s5b and redirect
>
> > all
> > > the changes to s5. All comments are addressed. Chain recipe is added to
> > > address #1270 <https://github.com/kaldi-asr/kaldi/issues/1270>
> > >
> > > —
> > > You are receiving this because you were mentioned.
>
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269441818>,
>
> > or mute
> > > the thread
> > > <https://github.com/notifications/unsubscribe-auth/
> > AKisX1PpyoTWPTkmBdI5FZFw1MQrrs0Yks5rMhkwgaJpZM4LSi8m>
> > > .
> > >
> >
> > —
> > You are receiving this because you were mentioned.
>
> > Reply to this email directly, view it on GitHub
> > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269449318>,
> or
> mute
> > the thread
> > <https://github.com/notifications/unsubscribe-auth/
> ADJVux9yfp0OR2IyKAlhJ7Md3sGDSf9tks5rMij0gaJpZM4LSi8m>
> > .
> >
>
> —
> You are receiving this because you were mentioned.
>
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269522522>, or
> mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/
> AKisX4ROHXQe6HYxw91cXkA1pPONxzG6ks5rMrB2gaJpZM4LSi8m>
> .
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269527183>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu-urAbJ0DqN-DzMT1nhIiZTjH48wks5rMrgigaJpZM4LSi8m>
> .
>

naxingyu commented:
I’ve already changed the PR to replace s5… It’s easy to create a different recipe (s5b) if it is the final decision…

X.

> 在 2016年12月29日，03:49，Daniel Povey <notifications@github.com> 写道：
> 
> OK.. Xingyu-- if it's easy to go back to creating a different recipe (s5b)
> then go ahead... I think you already changed the PR...
> 
> 
> On Wed, Dec 28, 2016 at 11:27 AM, jtrmal <notifications@github.com> wrote:
> 
> > Yes, I meant creating s5b, sorry.
> > I was thinking it might be beneficial because of changes in the scoring
> > scripts and and data preprocessing, I. E. Loss in the
> > continuity/comparability of the scores.
> > Y.
> >
> >
> > On Dec 28, 2016 7:54 PM, "Daniel Povey" <notifications@github.com> wrote:
> >
> > @jtrmal, can you clarify about "creating a new recipe", do you mean moving
> > to s5b? If so, what do you think the advantages are of having a new recipe
> > number?
> > I'm not sure that I want to encourage too much of this duplication in too
> > many of the recipes, it will bulk up the checked-out repo.
> >
> >
> >
> >
> > On Wed, Dec 28, 2016 at 1:16 AM, jtrmal <notifications@github.com> wrote:
> >
> > > I agree that creating a new recipe is better option.
> > > What was causing sore eyes in the original recipe was the text data
> > > preprocessing. That script was huge and I had terrible times just trying
> > to
> > > fix things to work for me. And I wasn;t trying to improve it -- just
> > trying
> > > to make it work.
> > > So perhaps it might be beneficial to make the script perhaps more verbose
> > > or structured differently or some of the functionality moved into a
> > > separate script?
> > > y.
> > >
> > > On Wed, Dec 28, 2016 at 9:09 AM, Xingyu Na <notifications@github.com>
> > > wrote:
> > >
> > > > @danpovey <https://github.com/danpovey> I've removed s5b and redirect
> >
> > > all
> > > > the changes to s5. All comments are addressed. Chain recipe is added to
> > > > address #1270 <https://github.com/kaldi-asr/kaldi/issues/1270>
> > > >
> > > > —
> > > > You are receiving this because you were mentioned.
> >
> > > > Reply to this email directly, view it on GitHub
> > > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269441818>,
> >
> > > or mute
> > > > the thread
> > > > <https://github.com/notifications/unsubscribe-auth/
> > > AKisX1PpyoTWPTkmBdI5FZFw1MQrrs0Yks5rMhkwgaJpZM4LSi8m>
> > > > .
> > > >
> > >
> > > —
> > > You are receiving this because you were mentioned.
> >
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269449318>,
> > or
> > mute
> > > the thread
> > > <https://github.com/notifications/unsubscribe-auth/
> > ADJVux9yfp0OR2IyKAlhJ7Md3sGDSf9tks5rMij0gaJpZM4LSi8m>
> > > .
> > >
> >
> > —
> > You are receiving this because you were mentioned.
> >
> > Reply to this email directly, view it on GitHub
> > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269522522>, or
> > mute
> > the thread
> > <https://github.com/notifications/unsubscribe-auth/
> > AKisX4ROHXQe6HYxw91cXkA1pPONxzG6ks5rMrB2gaJpZM4LSi8m>
> > .
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269527183>, or mute
> > the thread
> > <https://github.com/notifications/unsubscribe-auth/ADJVu-urAbJ0DqN-DzMT1nhIiZTjH48wks5rMrgigaJpZM4LSi8m>
> > .
> >
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269530548>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ADKpxMlad8-_noiPpElyv_1bj4rLnngBks5rMr0xgaJpZM4LSi8m>.
>

danpovey commented:
It's OK, I don't want to create any hassle, we can just replace the old
recipe and warn people to be careful.


On Wed, Dec 28, 2016 at 5:50 PM, Xingyu Na <notifications@github.com> wrote:

> I’ve already changed the PR to replace s5… It’s easy to create a different
> recipe (s5b) if it is the final decision…
>
> X.
>
>
> > 在 2016年12月29日，03:49，Daniel Povey <notifications@github.com> 写道：
> >
> > OK.. Xingyu-- if it's easy to go back to creating a different recipe
> (s5b)
> > then go ahead... I think you already changed the PR...
> >
> >
> > On Wed, Dec 28, 2016 at 11:27 AM, jtrmal <notifications@github.com>
> wrote:
> >
> > > Yes, I meant creating s5b, sorry.
> > > I was thinking it might be beneficial because of changes in the scoring
> > > scripts and and data preprocessing, I. E. Loss in the
> > > continuity/comparability of the scores.
> > > Y.
> > >
> > >
> > > On Dec 28, 2016 7:54 PM, "Daniel Povey" <notifications@github.com>
> wrote:
> > >
> > > @jtrmal, can you clarify about "creating a new recipe", do you mean
> moving
> > > to s5b? If so, what do you think the advantages are of having a new
> recipe
> > > number?
> > > I'm not sure that I want to encourage too much of this duplication in
> too
> > > many of the recipes, it will bulk up the checked-out repo.
> > >
> > >
> > >
> > >
> > > On Wed, Dec 28, 2016 at 1:16 AM, jtrmal <notifications@github.com>
> wrote:
> > >
> > > > I agree that creating a new recipe is better option.
> > > > What was causing sore eyes in the original recipe was the text data
> > > > preprocessing. That script was huge and I had terrible times just
> trying
> > > to
> > > > fix things to work for me. And I wasn;t trying to improve it -- just
> > > trying
> > > > to make it work.
> > > > So perhaps it might be beneficial to make the script perhaps more
> verbose
> > > > or structured differently or some of the functionality moved into a
> > > > separate script?
> > > > y.
> > > >
> > > > On Wed, Dec 28, 2016 at 9:09 AM, Xingyu Na <notifications@github.com
> >
> > > > wrote:
> > > >
> > > > > @danpovey <https://github.com/danpovey> I've removed s5b and
> redirect
> > >
> > > > all
> > > > > the changes to s5. All comments are addressed. Chain recipe is
> added to
> > > > > address #1270 <https://github.com/kaldi-asr/kaldi/issues/1270>
> > > > >
> > > > > —
> > > > > You are receiving this because you were mentioned.
> > >
> > > > > Reply to this email directly, view it on GitHub
> > > > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-
> 269441818>,
> > >
> > > > or mute
> > > > > the thread
> > > > > <https://github.com/notifications/unsubscribe-auth/
> > > > AKisX1PpyoTWPTkmBdI5FZFw1MQrrs0Yks5rMhkwgaJpZM4LSi8m>
> > > > > .
> > > > >
> > > >
> > > > —
> > > > You are receiving this because you were mentioned.
> > >
> > > > Reply to this email directly, view it on GitHub
> > > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269449318
> >,
> > > or
> > > mute
> > > > the thread
> > > > <https://github.com/notifications/unsubscribe-auth/
> > > ADJVux9yfp0OR2IyKAlhJ7Md3sGDSf9tks5rMij0gaJpZM4LSi8m>
> > > > .
> > > >
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > >
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269522522>,
> or
> > > mute
> > > the thread
> > > <https://github.com/notifications/unsubscribe-auth/
> > > AKisX4ROHXQe6HYxw91cXkA1pPONxzG6ks5rMrB2gaJpZM4LSi8m>
> > > .
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269527183>,
> or mute
> > > the thread
> > > <https://github.com/notifications/unsubscribe-auth/ADJVu-urAbJ0DqN-
> DzMT1nhIiZTjH48wks5rMrgigaJpZM4LSi8m>
> > > .
> > >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269530548>, or
> mute the thread <https://github.com/notifications/unsubscribe-
> auth/ADKpxMlad8-_noiPpElyv_1bj4rLnngBks5rMr0xgaJpZM4LSi8m>.
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269569976>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu1lXcyGVEEEWH1eUvY_1BZhWpOUEks5rMxHcgaJpZM4LSi8m>
> .
>

naxingyu commented:
There is no hassle :)
Anyway, replace or create, I think this PR ready for another review. I’ve created two installation scripts for mmseg and kaldi_lm, so now we won’t install anything in the recipe.

X.

> 在 2016年12月29日，10:46，Daniel Povey <notifications@github.com> 写道：
> 
> It's OK, I don't want to create any hassle, we can just replace the old
> recipe and warn people to be careful.
> 
> 
> On Wed, Dec 28, 2016 at 5:50 PM, Xingyu Na <notifications@github.com> wrote:
> 
> > I’ve already changed the PR to replace s5… It’s easy to create a different
> > recipe (s5b) if it is the final decision…
> >
> > X.
> >
> >
> > > 在 2016年12月29日，03:49，Daniel Povey <notifications@github.com> 写道：
> > >
> > > OK.. Xingyu-- if it's easy to go back to creating a different recipe
> > (s5b)
> > > then go ahead... I think you already changed the PR...
> > >
> > >
> > > On Wed, Dec 28, 2016 at 11:27 AM, jtrmal <notifications@github.com>
> > wrote:
> > >
> > > > Yes, I meant creating s5b, sorry.
> > > > I was thinking it might be beneficial because of changes in the scoring
> > > > scripts and and data preprocessing, I. E. Loss in the
> > > > continuity/comparability of the scores.
> > > > Y.
> > > >
> > > >
> > > > On Dec 28, 2016 7:54 PM, "Daniel Povey" <notifications@github.com>
> > wrote:
> > > >
> > > > @jtrmal, can you clarify about "creating a new recipe", do you mean
> > moving
> > > > to s5b? If so, what do you think the advantages are of having a new
> > recipe
> > > > number?
> > > > I'm not sure that I want to encourage too much of this duplication in
> > too
> > > > many of the recipes, it will bulk up the checked-out repo.
> > > >
> > > >
> > > >
> > > >
> > > > On Wed, Dec 28, 2016 at 1:16 AM, jtrmal <notifications@github.com>
> > wrote:
> > > >
> > > > > I agree that creating a new recipe is better option.
> > > > > What was causing sore eyes in the original recipe was the text data
> > > > > preprocessing. That script was huge and I had terrible times just
> > trying
> > > > to
> > > > > fix things to work for me. And I wasn;t trying to improve it -- just
> > > > trying
> > > > > to make it work.
> > > > > So perhaps it might be beneficial to make the script perhaps more
> > verbose
> > > > > or structured differently or some of the functionality moved into a
> > > > > separate script?
> > > > > y.
> > > > >
> > > > > On Wed, Dec 28, 2016 at 9:09 AM, Xingyu Na <notifications@github.com
> > >
> > > > > wrote:
> > > > >
> > > > > > @danpovey <https://github.com/danpovey> I've removed s5b and
> > redirect
> > > >
> > > > > all
> > > > > > the changes to s5. All comments are addressed. Chain recipe is
> > added to
> > > > > > address #1270 <https://github.com/kaldi-asr/kaldi/issues/1270>
> > > > > >
> > > > > > —
> > > > > > You are receiving this because you were mentioned.
> > > >
> > > > > > Reply to this email directly, view it on GitHub
> > > > > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-
> > 269441818>,
> > > >
> > > > > or mute
> > > > > > the thread
> > > > > > <https://github.com/notifications/unsubscribe-auth/
> > > > > AKisX1PpyoTWPTkmBdI5FZFw1MQrrs0Yks5rMhkwgaJpZM4LSi8m>
> > > > > > .
> > > > > >
> > > > >
> > > > > —
> > > > > You are receiving this because you were mentioned.
> > > >
> > > > > Reply to this email directly, view it on GitHub
> > > > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269449318
> > >,
> > > > or
> > > > mute
> > > > > the thread
> > > > > <https://github.com/notifications/unsubscribe-auth/
> > > > ADJVux9yfp0OR2IyKAlhJ7Md3sGDSf9tks5rMij0gaJpZM4LSi8m>
> > > > > .
> > > > >
> > > >
> > > > —
> > > > You are receiving this because you were mentioned.
> > > >
> > > > Reply to this email directly, view it on GitHub
> > > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269522522>,
> > or
> > > > mute
> > > > the thread
> > > > <https://github.com/notifications/unsubscribe-auth/
> > > > AKisX4ROHXQe6HYxw91cXkA1pPONxzG6ks5rMrB2gaJpZM4LSi8m>
> > > > .
> > > >
> > > > —
> > > > You are receiving this because you were mentioned.
> > > > Reply to this email directly, view it on GitHub
> > > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269527183>,
> > or mute
> > > > the thread
> > > > <https://github.com/notifications/unsubscribe-auth/ADJVu-urAbJ0DqN-
> > DzMT1nhIiZTjH48wks5rMrgigaJpZM4LSi8m>
> > > > .
> > > >
> > > —
> > > You are receiving this because you authored the thread.
> > > Reply to this email directly, view it on GitHub <
> > https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269530548>, or
> > mute the thread <https://github.com/notifications/unsubscribe-
> > auth/ADKpxMlad8-_noiPpElyv_1bj4rLnngBks5rMr0xgaJpZM4LSi8m>.
> > >
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269569976>, or mute
> > the thread
> > <https://github.com/notifications/unsubscribe-auth/ADJVu1lXcyGVEEEWH1eUvY_1BZhWpOUEks5rMxHcgaJpZM4LSi8m>
> > .
> >
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269574198>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ADKpxC_ru37ucWb2bjaqUc-EgIWiCIjMks5rMx8IgaJpZM4LSi8m>.
>

danpovey commented:
That's great.
I'll leave the rest up to @jtrmal-- Yenda, you decide on the question of
replace vs. change.


On Wed, Dec 28, 2016 at 6:54 PM, Xingyu Na <notifications@github.com> wrote:

> There is no hassle :)
> Anyway, replace or create, I think this PR ready for another review. I’ve
> created two installation scripts for mmseg and kaldi_lm, so now we won’t
> install anything in the recipe.
>
> X.
>
>
> > 在 2016年12月29日，10:46，Daniel Povey <notifications@github.com> 写道：
> >
> > It's OK, I don't want to create any hassle, we can just replace the old
> > recipe and warn people to be careful.
> >
> >
> > On Wed, Dec 28, 2016 at 5:50 PM, Xingyu Na <notifications@github.com>
> wrote:
> >
> > > I’ve already changed the PR to replace s5… It’s easy to create a
> different
> > > recipe (s5b) if it is the final decision…
> > >
> > > X.
> > >
> > >
> > > > 在 2016年12月29日，03:49，Daniel Povey <notifications@github.com> 写道：
> > > >
> > > > OK.. Xingyu-- if it's easy to go back to creating a different recipe
> > > (s5b)
> > > > then go ahead... I think you already changed the PR...
> > > >
> > > >
> > > > On Wed, Dec 28, 2016 at 11:27 AM, jtrmal <notifications@github.com>
> > > wrote:
> > > >
> > > > > Yes, I meant creating s5b, sorry.
> > > > > I was thinking it might be beneficial because of changes in the
> scoring
> > > > > scripts and and data preprocessing, I. E. Loss in the
> > > > > continuity/comparability of the scores.
> > > > > Y.
> > > > >
> > > > >
> > > > > On Dec 28, 2016 7:54 PM, "Daniel Povey" <notifications@github.com>
> > > wrote:
> > > > >
> > > > > @jtrmal, can you clarify about "creating a new recipe", do you mean
> > > moving
> > > > > to s5b? If so, what do you think the advantages are of having a new
> > > recipe
> > > > > number?
> > > > > I'm not sure that I want to encourage too much of this duplication
> in
> > > too
> > > > > many of the recipes, it will bulk up the checked-out repo.
> > > > >
> > > > >
> > > > >
> > > > >
> > > > > On Wed, Dec 28, 2016 at 1:16 AM, jtrmal <notifications@github.com>
> > > wrote:
> > > > >
> > > > > > I agree that creating a new recipe is better option.
> > > > > > What was causing sore eyes in the original recipe was the text
> data
> > > > > > preprocessing. That script was huge and I had terrible times just
> > > trying
> > > > > to
> > > > > > fix things to work for me. And I wasn;t trying to improve it --
> just
> > > > > trying
> > > > > > to make it work.
> > > > > > So perhaps it might be beneficial to make the script perhaps more
> > > verbose
> > > > > > or structured differently or some of the functionality moved
> into a
> > > > > > separate script?
> > > > > > y.
> > > > > >
> > > > > > On Wed, Dec 28, 2016 at 9:09 AM, Xingyu Na <
> notifications@github.com
> > > >
> > > > > > wrote:
> > > > > >
> > > > > > > @danpovey <https://github.com/danpovey> I've removed s5b and
> > > redirect
> > > > >
> > > > > > all
> > > > > > > the changes to s5. All comments are addressed. Chain recipe is
> > > added to
> > > > > > > address #1270 <https://github.com/kaldi-asr/kaldi/issues/1270>
> > > > > > >
> > > > > > > —
> > > > > > > You are receiving this because you were mentioned.
> > > > >
> > > > > > > Reply to this email directly, view it on GitHub
> > > > > > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-
>
> > > 269441818>,
> > > > >
> > > > > > or mute
> > > > > > > the thread
> > > > > > > <https://github.com/notifications/unsubscribe-auth/
> > > > > > AKisX1PpyoTWPTkmBdI5FZFw1MQrrs0Yks5rMhkwgaJpZM4LSi8m>
> > > > > > > .
> > > > > > >
> > > > > >
> > > > > > —
> > > > > > You are receiving this because you were mentioned.
> > > > >
> > > > > > Reply to this email directly, view it on GitHub
> > > > > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-
> 269449318
> > > >,
> > > > > or
> > > > > mute
> > > > > > the thread
> > > > > > <https://github.com/notifications/unsubscribe-auth/
> > > > > ADJVux9yfp0OR2IyKAlhJ7Md3sGDSf9tks5rMij0gaJpZM4LSi8m>
> > > > > > .
> > > > > >
> > > > >
> > > > > —
> > > > > You are receiving this because you were mentioned.
> > > > >
> > > > > Reply to this email directly, view it on GitHub
> > > > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-
> 269522522>,
> > > or
> > > > > mute
> > > > > the thread
> > > > > <https://github.com/notifications/unsubscribe-auth/
> > > > > AKisX4ROHXQe6HYxw91cXkA1pPONxzG6ks5rMrB2gaJpZM4LSi8m>
> > > > > .
> > > > >
> > > > > —
> > > > > You are receiving this because you were mentioned.
> > > > > Reply to this email directly, view it on GitHub
> > > > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-
> 269527183>,
> > > or mute
> > > > > the thread
> > > > > <https://github.com/notifications/unsubscribe-
> auth/ADJVu-urAbJ0DqN-
> > > DzMT1nhIiZTjH48wks5rMrgigaJpZM4LSi8m>
> > > > > .
> > > > >
> > > > —
> > > > You are receiving this because you authored the thread.
> > > > Reply to this email directly, view it on GitHub <
> > > https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269530548>,
> or
> > > mute the thread <https://github.com/notifications/unsubscribe-
> > > auth/ADKpxMlad8-_noiPpElyv_1bj4rLnngBks5rMr0xgaJpZM4LSi8m>.
> > > >
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269569976>,
> or mute
> > > the thread
> > > <https://github.com/notifications/unsubscribe-
> auth/ADJVu1lXcyGVEEEWH1eUvY_1BZhWpOUEks5rMxHcgaJpZM4LSi8m>
> > > .
> > >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269574198>, or
> mute the thread <https://github.com/notifications/unsubscribe-
> auth/ADKpxC_ru37ucWb2bjaqUc-EgIWiCIjMks5rMx8IgaJpZM4LSi8m>.
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269574751>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu9z3OgSdLAcMz6gyr_ZmehnUxmgKks5rMyEDgaJpZM4LSi8m>
> .
>

jtrmal commented:
Ok,  I was thinking about it and perhaps it's better to go with the s5, to
prevent unmaintained recipes.  I think we never imply the recipe won't
change substantially and the experience from babel is that maintaining the
older versions is just too much hassle.  So Xingyu,  let's just keep it as
it is.
I will try to send review today.

On Dec 29, 2016 3:59 AM, "Daniel Povey" <notifications@github.com> wrote:

That's great.
I'll leave the rest up to @jtrmal-- Yenda, you decide on the question of
replace vs. change.



On Wed, Dec 28, 2016 at 6:54 PM, Xingyu Na <notifications@github.com> wrote:

> There is no hassle :)
> Anyway, replace or create, I think this PR ready for another review. I’ve
> created two installation scripts for mmseg and kaldi_lm, so now we won’t
> install anything in the recipe.
>
> X.
>
>
> > 在 2016年12月29日，10:46，Daniel Povey <notifications@github.com> 写道：
> >
> > It's OK, I don't want to create any hassle, we can just replace the old
> > recipe and warn people to be careful.
> >
> >
> > On Wed, Dec 28, 2016 at 5:50 PM, Xingyu Na <notifications@github.com>
> wrote:
> >
> > > I’ve already changed the PR to replace s5… It’s easy to create a
> different
> > > recipe (s5b) if it is the final decision…
> > >
> > > X.
> > >
> > >
> > > > 在 2016年12月29日，03:49，Daniel Povey <notifications@github.com> 写道：
> > > >
> > > > OK.. Xingyu-- if it's easy to go back to creating a different recipe
> > > (s5b)
> > > > then go ahead... I think you already changed the PR...
> > > >
> > > >
> > > > On Wed, Dec 28, 2016 at 11:27 AM, jtrmal <notifications@github.com>
> > > wrote:
> > > >
> > > > > Yes, I meant creating s5b, sorry.
> > > > > I was thinking it might be beneficial because of changes in the
> scoring
> > > > > scripts and and data preprocessing, I. E. Loss in the
> > > > > continuity/comparability of the scores.
> > > > > Y.
> > > > >
> > > > >
> > > > > On Dec 28, 2016 7:54 PM, "Daniel Povey" <notifications@github.com>
> > > wrote:
> > > > >
> > > > > @jtrmal, can you clarify about "creating a new recipe", do you
mean
> > > moving
> > > > > to s5b? If so, what do you think the advantages are of having a
new
> > > recipe
> > > > > number?
> > > > > I'm not sure that I want to encourage too much of this duplication
> in
> > > too
> > > > > many of the recipes, it will bulk up the checked-out repo.
> > > > >
> > > > >
> > > > >
> > > > >
> > > > > On Wed, Dec 28, 2016 at 1:16 AM, jtrmal <notifications@github.com>
> > > wrote:
> > > > >
> > > > > > I agree that creating a new recipe is better option.
> > > > > > What was causing sore eyes in the original recipe was the text
> data
> > > > > > preprocessing. That script was huge and I had terrible times
just
> > > trying
> > > > > to
> > > > > > fix things to work for me. And I wasn;t trying to improve it --
> just
> > > > > trying
> > > > > > to make it work.
> > > > > > So perhaps it might be beneficial to make the script perhaps
more
> > > verbose
> > > > > > or structured differently or some of the functionality moved
> into a
> > > > > > separate script?
> > > > > > y.
> > > > > >
> > > > > > On Wed, Dec 28, 2016 at 9:09 AM, Xingyu Na <
> notifications@github.com
> > > >
> > > > > > wrote:
> > > > > >
> > > > > > > @danpovey <https://github.com/danpovey> I've removed s5b and
> > > redirect
> > > > >
> > > > > > all
> > > > > > > the changes to s5. All comments are addressed. Chain recipe is
> > > added to
> > > > > > > address #1270 <https://github.com/kaldi-asr/kaldi/issues/1270>
> > > > > > >
> > > > > > > —
> > > > > > > You are receiving this because you were mentioned.
> > > > >
> > > > > > > Reply to this email directly, view it on GitHub
> > > > > > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-
>
> > > 269441818>,
> > > > >
> > > > > > or mute
> > > > > > > the thread
> > > > > > > <https://github.com/notifications/unsubscribe-auth/
> > > > > > AKisX1PpyoTWPTkmBdI5FZFw1MQrrs0Yks5rMhkwgaJpZM4LSi8m>
> > > > > > > .
> > > > > > >
> > > > > >
> > > > > > —
> > > > > > You are receiving this because you were mentioned.

> > > > >
> > > > > > Reply to this email directly, view it on GitHub
> > > > > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-

> 269449318
> > > >,
> > > > > or
> > > > > mute
> > > > > > the thread
> > > > > > <https://github.com/notifications/unsubscribe-auth/
> > > > > ADJVux9yfp0OR2IyKAlhJ7Md3sGDSf9tks5rMij0gaJpZM4LSi8m>
> > > > > > .
> > > > > >
> > > > >
> > > > > —
> > > > > You are receiving this because you were mentioned.

> > > > >
> > > > > Reply to this email directly, view it on GitHub
> > > > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-

> 269522522>,
> > > or
> > > > > mute
> > > > > the thread
> > > > > <https://github.com/notifications/unsubscribe-auth/
> > > > > AKisX4ROHXQe6HYxw91cXkA1pPONxzG6ks5rMrB2gaJpZM4LSi8m>
> > > > > .
> > > > >
> > > > > —
> > > > > You are receiving this because you were mentioned.

> > > > > Reply to this email directly, view it on GitHub
> > > > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-
> 269527183>,

> > > or mute
> > > > > the thread
> > > > > <https://github.com/notifications/unsubscribe-
> auth/ADJVu-urAbJ0DqN-
> > > DzMT1nhIiZTjH48wks5rMrgigaJpZM4LSi8m>
> > > > > .
> > > > >
> > > > —
> > > > You are receiving this because you authored the thread.
> > > > Reply to this email directly, view it on GitHub <
> > > https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269530548>,
> or
> > > mute the thread <https://github.com/notifications/unsubscribe-
> > > auth/ADKpxMlad8-_noiPpElyv_1bj4rLnngBks5rMr0xgaJpZM4LSi8m>.
> > > >
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269569976>,
> or mute
> > > the thread
> > > <https://github.com/notifications/unsubscribe-
> auth/ADJVu1lXcyGVEEEWH1eUvY_1BZhWpOUEks5rMxHcgaJpZM4LSi8m>
> > > .
> > >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269574198>, or
> mute the thread <https://github.com/notifications/unsubscribe-
> auth/ADKpxC_ru37ucWb2bjaqUc-EgIWiCIjMks5rMx8IgaJpZM4LSi8m>.
> >
>
> —
> You are receiving this because you were mentioned.

> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269574751>, or
mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu9z3OgSdLAcMz6gyr_
ZmehnUxmgKks5rMyEDgaJpZM4LSi8m>
> .
>

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269575029>, or mute
the thread
<https://github.com/notifications/unsubscribe-auth/AKisX4_zQpdO1P4EIj8x7PwakujUMOhiks5rMyINgaJpZM4LSi8m>
.

naxingyu commented:
@jtrmal I commit the change to scoring scripts as #1281

naxingyu commented:
Chain online decoding is added. @jtrmal

jtrmal commented on egs/wsj/s5/utils/data/limit_feature_dim.sh:
```diff
@@ -0,0 +1,43 @@
+. utils/parse_options.sh
+
+if [ $# != 3 ]; then
+  echo "Usage: "
```
Brief (one-sentence) explanation please.

jtrmal commented on egs/hkust/s5/run.sh:
```diff
@@ -123,51 +113,25 @@ utils/mkgraph.sh data/lang_test exp/tri5a exp/tri5a/graph || exit 1;
-local/nnet2/run_convnet.sh
+# online nnet3
+local/nnet3/run_tdnn.sh
 
```
do you have the chain recipe? If so, pleas add into the run script as well

jtrmal commented on egs/hkust/s5/local/chain/run_tdnn.sh:
```diff
@@ -0,0 +1,209 @@
+#!/bin/bash
```
For the chain and nnet3 scripts, if you've copied them from another directory/eg, please mention the source on the top -- that would help tracking of changes (for example propagating changes when the originating script changes).

naxingyu commented:
Done.

> 在 2016年12月30日，15:51，jtrmal <notifications@github.com> 写道：
> 
> @jtrmal commented on this pull request.
> 
> In egs/wsj/s5/utils/data/limit_feature_dim.sh <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14739128>:
> 
> > @@ -0,0 +1,43 @@
> +#!/bin/bash
> +
> +# Copyright 2016  Alibaba Robotics Corp. (author: Xingyu Na)
> +# Apache 2.0
> +
> +. utils/parse_options.sh
> +
> +if [ $# != 3 ]; then
> +  echo "Usage: "
> Brief (one-sentence) explanation please.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14739128>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ADKpxJrCLJUqFgH0ttag-3jYwmPQ0hWCks5rNLfugaJpZM4LSi8m>.
>

jtrmal commented on egs/hkust/s5/local/hkust_data_prep.sh:
```diff
@@ -1,97 +1,74 @@
-  fi
+if [ ! `echo $PYTHONPATH | grep mmseg` ]; then
+  echo "mmseg is not found. Checkout tools/extra/install_mmseg.sh"
+  exit 1
```
Perhaps a less brittle check would be a simple oneliner that tries to import given module
something like (probably incorrect syntactically)
python -c "import sys; try: import mmseg; except: sys.exit(1);"

jtrmal replied:
The reason is that the user might have already mmseg installed as a normal python package.

jtrmal commented on egs/hkust/s5/local/nnet3/run_ivector_common.sh:
```diff
@@ -7,31 +7,25 @@
 
```
similarly as with the other scripts -- please mention the origin if you know it.

jtrmal commented on egs/wsj/s5/utils/data/limit_feature_dim.sh:
```diff
@@ -3,6 +3,9 @@
 
+# This script allows you to create a data directory from limited
+# dim range of the source data dir.
+
```
OK, but I'd prefer adding it also to the message that gets printed out when incorrect number of parameters.
Also, perhaps the wording might be more explanatory -- something like 

The script creates a new data directory by selecting/copying a specified dimension range of the features in the source directory. 
Or something to that extent..

jtrmal commented on egs/hkust/s5/RESULTS:
```diff
@@ -18,11 +17,26 @@ exp/cnn5c/decode/cer_10:%WER 40.13 [ 22536 / 56154, 2329 ins, 3962 del, 16245 su
+
+# nnet3 mfcc results (using speed perturbed data)
 exp/nnet3/tdnn_sp/decode_dev/cer_10:%WER 33.79 [ 18977 / 56154, 2027 ins, 3485 del, 13465 sub ]
-exp/nnet3/lstm_sp_ld5/decode_dev/cer_9:%WER 33.51 [ 18815 / 56154, 1813 ins, 3249 del, 13753 sub ]
```
the lstm_sp_ld5 and tdnn_sp (on the left side of the diff) were obtained before or after adding the pitch?

naxingyu commented:
The results are before adding pitch. I added
# nnet3 mfcc results
In the comments to distinguish.

> 在 2016年12月30日，16:16，jtrmal <notifications@github.com> 写道：
> 
> @jtrmal commented on this pull request.
> 
> In egs/hkust/s5/RESULTS <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14740123>:
> 
> >  exp/nnet3/tdnn_sp/decode_dev/cer_10:%WER 33.79 [ 18977 / 56154, 2027 ins, 3485 del, 13465 sub ]
> -exp/nnet3/lstm_sp_ld5/decode_dev/cer_9:%WER 33.51 [ 18815 / 56154, 1813 ins, 3249 del, 13753 sub ]
> the lstm_sp_ld5 and tdnn_sp (on the left side of the diff) were obtained before or after adding the pitch?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14740123>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ADKpxDE-VRaxfhArOjzRHJwJGTgdS0Wjks5rNL3agaJpZM4LSi8m>.
>

naxingyu commented:
I commit a clearer RESULTS file, along with the addressing of all your comments.

X.

> 在 2016年12月30日，16:16，jtrmal <notifications@github.com> 写道：
> 
> @jtrmal commented on this pull request.
> 
> In egs/hkust/s5/RESULTS <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14740123>:
> 
> >  exp/nnet3/tdnn_sp/decode_dev/cer_10:%WER 33.79 [ 18977 / 56154, 2027 ins, 3485 del, 13465 sub ]
> -exp/nnet3/lstm_sp_ld5/decode_dev/cer_9:%WER 33.51 [ 18815 / 56154, 1813 ins, 3249 del, 13753 sub ]
> the lstm_sp_ld5 and tdnn_sp (on the left side of the diff) were obtained before or after adding the pitch?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14740123>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ADKpxDE-VRaxfhArOjzRHJwJGTgdS0Wjks5rNL3agaJpZM4LSi8m>.
>

naxingyu commented:
Wait, it seems I cannot compress a ’try/except’ block into a python one-liner. I need to reverse this change.

> 在 2016年12月30日，16:42，Xingyu Na <asr.naxingyu@gmail.com> 写道：
> 
> I commit a clearer RESULTS file, along with the addressing of all your comments.
> 
> X.
> 
>> 在 2016年12月30日，16:16，jtrmal <notifications@github.com <mailto:notifications@github.com>> 写道：
>> 
>> @jtrmal commented on this pull request.
>> 
>> In egs/hkust/s5/RESULTS <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14740123>:
>> 
>> >  exp/nnet3/tdnn_sp/decode_dev/cer_10:%WER 33.79 [ 18977 / 56154, 2027 ins, 3485 del, 13465 sub ]
>> -exp/nnet3/lstm_sp_ld5/decode_dev/cer_9:%WER 33.51 [ 18815 / 56154, 1813 ins, 3249 del, 13753 sub ]
>> the lstm_sp_ld5 and tdnn_sp (on the left side of the diff) were obtained before or after adding the pitch?
>> 
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14740123>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ADKpxDE-VRaxfhArOjzRHJwJGTgdS0Wjks5rNL3agaJpZM4LSi8m>.
>> 
>

jtrmal commented on egs/hkust/s5/local/hkust_data_prep.sh:
```diff
@@ -1,98 +1,73 @@
-if [ ! -d $HKUST_AUDIO_DIR ] || [ ! -d $HKUST_TEXT_DIR ]; then
+if [ ! -d $hkust_audio_dir ] || [ ! -d $hkust_text_dir ]; then
   echo "Error: run.sh requires two directory arguments"
   exit 1;
```
this should be local/hkust_data_prep.sh or $0, not run.sh

jtrmal commented on egs/hkust/s5/local/hkust_segment.py:
```diff
@@ -12,4 +12,4 @@
     for j in seg_txt(blks[i]):
       out_line += " " + j
-  print out_line     
+  print out_line
```
just noticed this -- please use 
from __future__ import print_function
and then

print(out_line)


that should be more opened towards python3 transition

naxingyu replied:
Resolved.

jtrmal commented on egs/wsj/s5/utils/data/limit_feature_dim.sh:
```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
```
@danpovey , perhaps you wanna have a look at this as it is going to utils/ dir?

danpovey commented on egs/wsj/s5/utils/data/limit_feature_dim.sh:
```diff
@@ -0,0 +1,48 @@
+  echo "The script creates a new data directory by selecting a specified"
+  echo "dimension range of the features in the source directory."
+  echo "e.g.:"
+  echo " $0 0:40 data/train_hires_pitch data/train_hires"
```
the example range should be 0:39 (my mistake probably)

danpovey commented on egs/wsj/s5/utils/data/limit_feature_dim.sh:
```diff
@@ -0,0 +1,48 @@
+mkdir -p $destdir
+utils/copy_data_dir.sh $srcdir $destdir
+
+if [ -f $destdir/cmvn.scp ]; then
```
otherwise the script is fine.

naxingyu commented:
It's been modified.


在 2016/12/31 5:34, Daniel Povey 写道:
>
> *@danpovey* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In egs/wsj/s5/utils/data/limit_feature_dim.sh 
> <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14790855>:
>
> > +
> +# Copyright 2016  Alibaba Robotics Corp. (author: Xingyu Na)
> +# Apache 2.0
> +
> +# The script creates a new data directory by selecting a specified
> +# dimension range of the features in the source directory.
> +
> +. utils/parse_options.sh
> +
> +if [ $# != 3 ]; then
> +  echo "Usage: "
> +  echo "  $0 <feat-dim-range> <srcdir> <destdir>"
> +  echo "The script creates a new data directory by selecting a specified"
> +  echo "dimension range of the features in the source directory."
> +  echo "e.g.:"
> +  echo " $0 0:40 data/train_hires_pitch data/train_hires"
>
> the example range should be 0:39 (my mistake probably)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/kaldi-asr/kaldi/pull/1280#pullrequestreview-14790855>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ADKpxH0YRV1ofdnIvk_vG7qGI16mmf4Rks5rNXjigaJpZM4LSi8m>.
>

jtrmal commented:
I think this is ready to be merged.
I think the text data preparation script could be reworked and perhaps some parts shared with gale, but there is  not immediate need.

naxingyu commented:
Do you mean the text prep or the dict prep script?
In both scripts, there are some perl expressions that can be shared. And the dict prep scripts of hkust and gale_mandarin should be the same.
What do you have in mind specifically?

X.

> 在 2017年1月2日，22:21，jtrmal <notifications@github.com> 写道：
> 
> I think this is ready to be merged.
> I think the text data preparation script could be reworked and perhaps some parts shared with gale, but there is not immediate need.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/kaldi-asr/kaldi/pull/1280#issuecomment-269978448>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ADKpxL1SKIWz32qjwo_hGe2FdIDuDVWBks5rOQfZgaJpZM4LSi8m>.
>
